.github/workflows/ci.yml

```diff
@@ -0,0 +1,139 @@
+name: CI
+
+on:
+    push:
+        branches:
+            - "main"
+            - "release/*"
+    pull_request:
+        branches:
+            - "main"
+            - "release/*"
+    release:
+        types: [created]
+
+jobs:
```

Comment:
Just curious, are there reasons that we want to keep these as separate jobs rather than steps?

Reply:
A few reasons I think (though maybe there are reasons to flatten the workflow into steps that I'm not aware of, please lmk!):

- For non-dependent tasks jobs can run in parallel, which can make the workflow run faster. Also for non-dependent tasks, jobs allow us to provide a clean environment for running each, which improves reproducability and could make it easier to understand whats going wrong in some failure scenarios.
- In the Actions failure reports, individual jobs' statuses will reported at the top level, which personally I like to be able to view at a glance. I like to be able to differentiate whether linting vs. type checking failed, for instance, without clicking in.
- Each job has its own log output, which, maybe this is a preference, but I think is nicer than one long dump of the workflow logs, especially as you add more tasks.
- Though we am not making use of this (yet), we could run specifically jobs conditionally. For example only running `build` on releases, as I think you might be suggesting below (and I think I agree).